code/gamesnd/gamesnd.cpp

```diff
@@ -703,22 +734,122 @@ bool required_string_no_create(const char* token, bool no_create)
     return true;
 }
 
+static GameSoundCycleType parse_cycle_type() {
+	if (optional_string("Sequential")) {
+		return GameSoundCycleType::Sequential;
+	} else if (optional_string("Random")) {
+		return GameSoundCycleType::Random;
+	} else {
+		error_display(0, "Failed to parse sound cycle type. Expected 'sequential' or 'random'. Got [%.32s]", next_tokens());
```

Comment:
I don't think `next_tokens()` consumes those tokens, so I'm pretty sure this will inevitably result in another parsing error, if somebody actually supplies an invalid cycle type.

Reply:
Good point. I added a `advance_to_eoln` call which should skip over the bad token in most cases.